src/backend/logger/logger.ts

```diff
@@ -355,10 +355,89 @@ export function logFileLocation(appName: string) {
   return join(gamesConfigPath, `${appName}.log`)
 }
 
+const logsWriters: { [appName: string]: LogWriter } = {}
+
+class LogWriter {
+  appName: string
+  queue: string[] | undefined
+  initialized: boolean
+  timeoutId: NodeJS.Timeout | undefined
+  filePath: string
+
+  constructor(appName: string) {
+    this.appName = appName
+    this.initialized = false
+    this.filePath = lastPlayLogFileLocation(appName)
+  }
+
+  logMessage(message: string) {
+    // initialize the value if undefined
+    if (this.queue === undefined) this.queue = []
+
+    // push messages to append to the log
+    this.queue.push(message)
+
+    // if the logger is initialized and we don't have a timeout,
+    // append the message and start a timeout
+    //
+    // otherwise it means there's a timeout already running that will
+    // write the elements in the queue in a second
+    if (this.initialized && !this.timeoutId) this.appendMessages()
+  }
+
+  async initLog() {
+    try {
+      const info = await getSystemInfo()
+      const systemInfo = await formatSystemInfo(info)
+
+      // init log file and then append message if any
+      writeFile(
+        this.filePath,
+        'System Info:\n' + `${systemInfo}\n` + '\n',
+        () => {
+          this.initialized = true
+          this.appendMessages()
+        }
+      )
+    } catch (error) {
+      logError(['Failed to fetch system information', error], LogPrefix.Backend)
+    }
+  }
+
+  appendMessages() {
+    const messagesToWrite = this.queue
+
+    // clear pending message if any
+    this.queue = []
+
+    // clear timeout if any
+    delete this.timeoutId
+
+    if (!messagesToWrite || !messagesToWrite.length) return
+
+    // if we have messages, write them and check again in 1 second
+    // we start the timeout before writing so we don't wait until
+    // the disk write
+    this.timeoutId = setTimeout(() => this.appendMessages(), 1000)
+
+    try {
+      appendFileSync(this.filePath, messagesToWrite.join(''))
+    } catch (error) {
+      // ignore failures if messages could not be written
+    }
+  }
+}
+
 export function appendGameLog(appName: string, message: string) {
-  appendFileSync(lastPlayLogFileLocation(appName), message)
+  if (!logsWriters[appName]) logsWriters[appName] = new LogWriter(appName)
```

Comment:
```suggestion
  logsWriters[appName] ??= new LogWriter(appName)
```

Reply:
I have to say that this new JS feature makes the code a lot more unreadable IMO.
is up to you guys to use it or not, I will try not to.

for me doing a `IF` is always easier to understand than using this `??=` 😅 😂

Reply:
I'd argue that a lot of "JS-isms" (specifically around null-checking "shortcuts" and functional programming) are hard to read when first confronted with them.
Nullish coalescing assignment *is* ever-so-slightly faster, but feel free to use whatever you wish to use

Reply:
at least for me I'm used to this syntax from other languages, I didn't know it was a thing in javascript

if it helps I can say you get used to it very quickly and then you miss that when it doesn't exist

Reply:
yeah, makes sense. Maybe its the age that is affecting me and making me stand for the 'old ways' 😂